clientd3d/statgame.c

```diff
@@ -370,13 +381,31 @@ void GameTimer(HWND hwnd, UINT id)
 /****************************************************************************/
 void GameIdle(void)
 {
+   static const std::chrono::duration<double> minInterval(1.0 / 60.0); // 1/60th of a second
+   static auto lastCallTime = std::chrono::high_resolution_clock::now();
+   auto currentTime = std::chrono::high_resolution_clock::now();
+   std::chrono::duration<double> timeSinceLastCall = currentTime - lastCallTime;
+   
+   // Guard to ensure the function is not called more than 60 times per second
+   if (timeSinceLastCall < minInterval) {
+       return;
+   }
+   
+   // Update the last call time
+   lastCallTime = currentTime;
    MoveUpdateServer();       // Update our position on server, if necessary
 
    AnimationTimerProc(hMain, 0);
    HandleKeys();
 
    // Are we the active foreground application?
    AnimationTimerAbort();
+
+   // If we're in the background, sleep to lower CPU load
+   if (GetWindowLong(GetActiveWindow(), GWLP_HINSTANCE) != (LONG_PTR)hInst)
```

Comment:
I can't tell whether this is the right check.  See e.g.
https://stackoverflow.com/questions/3940346/foreground-vs-active-window

If the app is not in the foreground, I think GetActiveWindow() is returning NULL, which probably "works" but only by accident.  Can you do some more research on this, and add error checking?